### PR TITLE
Typing Speed Test Game is Developed

### LINKED ISSUE #1362: [Add a Play]: Typing Speed Test Game on React JS
### What's the Play about?

**Describe the feature:**
A type speed testing game to calculate a user's typing speed in words per minute.

- User gets a paragraph that he has to type.
- Every correct word count positive for typing speed.
- Similarly, every wrong counts negative for typing speed.
- A timer for time elapsed.

Could you please assign me this issue?

### What ReactJS concept will be used to create this play?

**Feature will be developed using:** 

- Class based React JS.
- Use State hook of React.

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue
- [X] I am in Hacktoberfest

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by neelesh133**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

resolves #1362 

**Description**

A typing speed test game in which you have to write a given paragraph and will get your typing speed according to how many words you typed in how much time (words per minute).

**Features:**

- Type the highlighted words in the input field after starting the game.
- The correct words will turn green.
- The incorrect words will turn red.
- After typing the whole content user will get his final typing speed in wpm(words per minute).

# Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- The code follows all the standards and doesn't give prettier errors.
- The game is working fine on react-play website.
- The game has been tested for all conditions and is working fine.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

image
image
image
image

### CODE REVIEW

**Comment by neelesh133** [comment]
I have developed the typing speed test game. Could you please review it. Thank You! :)

File: src/plays/typingtestgame/Readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+- Language: js
+- Level: Intermediate
+
+## Creator Information
+
+- User: neelesh133
+- Gihub Link: https://github.com/neelesh133
+- Blog: 
+- Video:
```
Comment (priyankarpal) [inline src/plays/typingtestgame/Readme.md:15]: remove 
```suggestion

```

File: src/plays/typingtestgame/Readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+1. Typing_test.js
+   Single file with all the required code regarding the speed test box having a start button to start speed test and a meter to show current typing speed in wpm(words per minute) and timer to specify the time elapsed.
+
+## Consideration
+
+Update all considerations(if any)
+
+## Resources
+
+Update external resources(if any)
```
Comment (priyankarpal) [inline src/plays/typingtestgame/Readme.md:30]: please fill the section otherwise you can remove

File: src/plays/typingtestgame/cover.jpg
Comment (priyankarpal) [inline src/plays/typingtestgame/cover.jpg:1]: please use CDN link for the image

**Review by priyankarpal — CHANGES_REQUESTED** [review]

File: src/plays/typingtestgame/cover.jpg
Comment (neelesh133) [reply src/plays/typingtestgame/cover.jpg:1]: Could you explain how to use cdn link for the cover image please ?

Comment (priyankarpal) [reply src/plays/typingtestgame/cover.jpg:1]: use this method @neelesh133 

https://github.com/reactplay/react-play/assets/88102392/60c85cb3-3e32-4a5f-a4b9-c73ad625a8f0

Comment (neelesh133) [reply src/plays/typingtestgame/cover.jpg:1]: I mean cover.jpg is the image used for the cover image displayed on the page where all plays are displayed. Where should I paste the link in order to get the same output, or i should use cover.png instead? Please could you help me out there ? Thank you :)

Comment (priyankarpal) [reply src/plays/typingtestgame/cover.jpg:1]: ok I got it

**Review by priyankarpal — CHANGES_REQUESTED** [review]
you're using react so I will suggest to change extension name to `jsx`

**Comment by neelesh133** [comment]
> you're using react so I will suggest to change extension name to `jsx`

Sure! I have applied the required changes. :)

**Review by priyankarpal — APPROVED** [review]

File: src/plays/typingtestgame/components/styles.css [hunk 3]
```diff
@@ -0,0 +1,104 @@
+@import url("https://fonts.googleapis.com/css?family=Muli&display=swap");
```
Comment (atapas) [inline src/plays/typingtestgame/components/styles.css:1]: @neelesh133 Do not override tag selectors this way. Please read this

https://docs.reactplay.io/How-To-Guides/how-to-style-play

You need add a unique class name of ur choice and use that as a prefix for your styles.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/plays/typingtestgame/components/styles.css (hunk 3 above)
Comment (neelesh133) [reply src/plays/typingtestgame/components/styles.css:1]: I will surely resolve it in a while. :)

Comment (neelesh133) [reply src/plays/typingtestgame/components/styles.css:1]: Unfortunately, the project has somehow stopped working and giving me an error on "npm start". I have installed node_modules through "npm i --legacy-peer-deps" successfully. The screenshot of the error is below-

image

Can you please help me out there ?

Comment (priyankarpal) [reply src/plays/typingtestgame/components/styles.css:1]: Are you still facing an issue?

**Review by priyankarpal — CHANGES_REQUESTED** [review]

**Comment by priyankarpal** [comment]
Since there haven't been any updates, we can move forward with closing this PR.

**Pull request closed without merging**